raft/fsm.cc

```diff
@@ -183,6 +183,7 @@ void fsm::become_follower(server_id leader) {
 }
 
 void fsm::become_candidate(bool is_prevote, bool is_leadership_transfer) {
+    auto prev_leader = is_follower() ? follower_state().current_leader : server_id{};
```

Comment:
What if a node restarts in the follower (voter or non-voter) state, how does it know who the leader is?

Reply:
That's a good question, I'm not sure if it learns about the leader automatically!
In that case, a voter would simply start a new election and become a leader (but that would mean we re-elect a leader whenever a node restarts - that doesn't sound great), and a non-voter would simply not learn until something happens in the cluster (e.g. a new entry is added, so the leader will send append_entries)
cc @gleb-cloudius

Reply:
We have a prevote, so there will no be re-election each time a node restarts. But a node (voting or non voting) learns about a leader only from `append_request`, `install_snapshot` and `read_quorum` messages. Until it gets one of those it will not know who the leaser is. Of course the problem does not exists in the vanilla raft since `append_requests` are sent periodically. We deviated from it and have this problem now. I am not sure how to best solve it. We may have special message to solicit empty `append_message` from a leader that will be sent by a node after restart (to all nodes in the config). It may be specially crafted append reject message as well.

Reply:
> We have a prevote, so there will no be re-election each time a node restarts.
Well prevote does not prevent it if logs are the same. Sticky leadership would, but we disabled it.